src/ruby/integration/default_test.go

```diff
@@ -0,0 +1,120 @@
+package integration_test
+
+import (
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/cloudfoundry/switchblade"
+	"github.com/sclevine/spec"
+
+	. "github.com/cloudfoundry/switchblade/matchers"
+	. "github.com/onsi/gomega"
+)
+
+func testDefault(platform switchblade.Platform, fixtures string) func(*testing.T, spec.G, spec.S) {
+	return func(t *testing.T, context spec.G, it spec.S) {
+		var (
+			Expect     = NewWithT(t).Expect
+			Eventually = NewWithT(t).Eventually
+
+			name string
+		)
+
+		it.Before(func() {
+			var err error
+			name, err = switchblade.RandomName()
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		it.After(func() {
+			Expect(platform.Delete.Execute(name)).To(Succeed())
+		})
+
+		context("when the ruby version is specified in the app", func() {
+			it("builds and runs the app with the specified version", func() {
+				deployment, _, err := platform.Deploy.
+					Execute(name, filepath.Join(fixtures, "default", "specified_ruby_version"))
+				Expect(err).NotTo(HaveOccurred())
+
+				Eventually(deployment).Should(Serve(ContainSubstring("ruby 3.1.6")).WithEndpoint("/ruby"))
+			})
+		})
+
+		context("when the ruby version is not specified in the app", func() {
+			it("builds and runs the app with the default version", func() {
+				deployment, _, err := platform.Deploy.
+					Execute(name, filepath.Join(fixtures, "default", "unspecified_ruby_version"))
+				Expect(err).NotTo(HaveOccurred())
+
+				Eventually(deployment).Should(Serve(MatchRegexp(`ruby 3\.\d+\.\d+`)).WithEndpoint("/ruby"))
+			})
+		})
+
+		context("rails7", func() {
+			it("builds and runs the app", func() {
```

Comment:
Just an FYI - I don't know what's the default Disk/Memory, but we have set Disk and Memory for many of these apps (e.g. https://github.com/cloudfoundry/ruby-buildpack/blob/v1.10.17/src/ruby/integration/rails_7_test.go#L30) - just something to think in case they fail on CF. I'm not sure if we have implemented setting Memory/Disk on switchblade

Reply:
On switchblade the default limit is "unbounded". Because it's just a docker container, and docker defaults to no limits. Unlike on CF where the default limit is whatever the platform settings are.

So there's really no reason to implement memory/disk/CPU limits on switchblade, as all that it would enable is testing how apps fail when there's not enough resources. Which I don't think is a useful integration test.

Reply:
My comment was only about when this switchblady code runs on CF in the concourse CI

Reply:
oh sorry i misunderstood that.

yes, we can keep an eye out for apps failing to start on CF. Good catch 👍